### DISCUSSION TITLE
useEffect does not work in production build

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [vercel/next.js#22314] [Help]
None of the React hooks are working on the production build right now, but they're working in development. Any ideas?

For instance:

```tsx
// index.tsx

import { useEffect } from 'react'

export default function Home() {
  useEffect(() => {
    prompt('hello')
  })

  return <p>The homepage contents</p>
}
```

No prompt. I've also tried with `useCallback` and `useState`. In case there was an issue with the prompt, I also tried getting output with `console.log` and `console.error`.

No errors in any of the logs.

Interestingly, I cannot figure out how to replicate this when starting from `create-next-app`, and I'm stuck on where to continue debugging. 

### Setup

- next: 10.0.7
- Typescript
- pnpm workspaces

**Comment by Nick-Mazuk** [comment]
After several hours of debugging, I can confirm that this isn't a Next.js issue. It was with another dependency.

**Reply by AugustSaintFreytag** [reply]
Well, which one?!

**Reply by Nick-Mazuk** [reply]
> Well, which one?!

I should have included that, but I honestly don't remember any more. I don't remember how I solved it, either.

**Comment by cuxs** [comment]
I'm experiencing the same issue when importing a css..

I have a page using styles from antd, so I import the css:

```javascript
//...
import React, { useEffect, useState } from 'react';
import { useRouter } from 'next/router';
import 'antd/dist/antd.css';
import styles from './myComponent.module.scss';


const MyComponent = ()=>{
  const router = useRouter()
  useEffect(()=>alert('hi babe'))
  useEffect(() => {
    const token = getToken();
    if (!token) {
      router.replace('/acceso')
    }
  }, [])
return (
    <div className={styles.leads__container}>HI</div>
)

}
//...
```
nothing shows up after doing the build:

![image](https://user-images.githubusercontent.com/8927894/144055289-25269629-dad6-4384-beec-9f2b1be2da4e.png)


But in development:

![image](https://user-images.githubusercontent.com/8927894/144055598-e0c0946e-0a9b-42d2-957e-4c9bb84d2fbe.png)


After removing the css import, all works as expected...

I'm using 
-`next@10.2.3`
-`react@17.0.2`
-`react-dom@17.0.2`

**Reply by cuxs** [reply]
Bumping next version to 11.1.2 seems to solve the problem.

**Comment by devih-21** [comment]
try add import useRouter and it working  
```
import { useEffect } from 'react'
import { useRouter } from "next/router";

export default function Home() {
  useEffect(() => {
    prompt('hello')
  })

  return <p>The homepage contents</p>
}
```

**Comment by hamzadotdev** [comment]
remove .next folder from the root directory ,  remove the next.config.js with following content:

const isProd = process.env.NODE_ENV === 'production'

module.exports = {
  assetPrefix: isProd ? '/fractumscrypt/' : '',
  images: {
    unoptimized: true,
  },
}

After removing the file everything started to work and execute npm run build command , hope so it will works
